utilities/pytest_utils.py

```
@@ -609,3 +610,49 @@ def update_cpu_arch_related_config(cpu_arch_option: str) -> None:
+            and getattr(getattr(cls, name), "__test__", True)
+            and not _is_xfail_no_run(getattr(cls, name))
+        ]
+        missing = [name for name in all_tests if name not in set(collected_names)]
+        if missing:
+            errors.append(f"{cls.__qualname__}: not collected: {missing}")
+
+    if errors:
+        raise pytest.UsageError("Incremental classes partially collected:\n" + "\n".join(errors))
+
+
+def _is_xfail_no_run(method: Any) -> bool:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `method`

(ANN401)